Core/RadioHandler.cpp

```diff
@@ -74,27 +68,40 @@ RadioHandlerClass::RadioHandlerClass() :
 	inputbuffer.setBlockSize(transferSamples);
 
 	stateFineTune = new shift_limited_unroll_C_sse_data_t();
+	adcrate = adcnominalfreq;
+	fprintf(stderr,"RadioHandlerClass::RadioHandlerClass\n");
 }
 
 RadioHandlerClass::~RadioHandlerClass()
 {
 	delete stateFineTune;
+	delete r2iqCntrl;
+	delete hardware;
+	delete fx3;
```

Comment:
I don't think this class owes the fx3 instance. it is allocated by someone else. better deallocate by the owner.

Reply:
That was freed by the owner while threads were still running, resulting in random crashes/freezes. That's why I did this change. I don't think, that it is better to deallocate by the owner. I think that it is better to use `std::shared_ptr`. Switching to `std::shared_ptr` may be the next thing to do. But incorrectly using smart pointer may introduce crazy bugs too...

Reply:
it should not happen at least for Windows. Can you point me the crash stacks of two threads? I can help debug.